File: C#/0021. Merge Two Sorted Lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode MergeTwoLists(ListNode l1, ListNode l2)
        {
            if (l1 == null)
            {
                return l2;
            }
            if (l2 == null)
            {
                return l1;
            }
            ListNode newhead = new ListNode(0);
            ListNode p = newhead;
            while(l1!=null && l2 != null)
            {
                if (l1.val <= l2.val)
                {
                    p.next = l1;
                    p = p.next;
                    l1 = l1.next;
                }
                else
                {
                    p.next = l2;
                    l2 = l2.next;
                    p = p.next;
                }
            }
            if (l1!=null)
            {
                p.next = l1;
            }
            if (l2 != null)
            {
                p.next = l2;
            }
            return newhead.next;
        }

}

File: 20200215Python-China/00 21. Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        res=ListNode(0)
        p=res
        while l1 and l2:
            if l1.val<l2.val:
                p.next=l1
                p=p.next
                l1=l1.next
            else:
                p.next=l2
                p=p.next
                l2=l2.next
        if l1:
            p.next=l1
        if l2:
            p.next=l2
        return res.next
        
